### ISSUE TITLE
About loss

### ISSUE THREAD

**Issue by SongJgit** [anubhabghosh/danse_jrnl#1]
In paper 
![](https://cdn.mathpix.com/snip/images/PLO7U4MOWdHNbrcUHDRVcKcSJEfvzOJsdmmPPFrUFzA.original.fullsize.png)
 
In supervised, should `n` be the dimension representing states? That is, `self.n_states`, But in the supervised code, the distribution of `X` is computed, but `n` is still the dimension of `Y`(self.n_obs).
https://github.com/anubhabghosh/danse_jrnl/blob/40fa6d6e9de44e7d0b31377c8110c5fa7b7252f8/src/danse_supervised.py#L127-L135

Alternatively, we could just write as 
```python
def compute_logpdf_Gaussian(self, X):
    
    _, T, n = X.shape 
    logprob = 0.5 * n * T * math.log(math.pi*2) - 0.5 * torch.logdet(self.L_xt_yt_current).sum(1) \
        - 0.5 * torch.einsum('nti,nti->nt',
        (X - self.mu_xt_yt_current), 
        torch.einsum('ntij,ntj->nti',torch.inverse(self.L_xt_yt_current), (X - self.mu_xt_yt_current))).sum(1)

    return logprob
```

**Comment by anubhabghosh** [comment]
Thanks for pointing this out, this doesn't matter in the optimization because it only affects the constant term. Addressed in the recent commit d36340a
